snuba/utils/streams/kafka.py

```diff
@@ -98,12 +171,50 @@ def poll(self, timeout: Optional[float] = None) -> Optional[KafkaMessage]:
             else:
                 raise ConsumerError(str(error))
 
-        return KafkaMessage(
+        result = KafkaMessage(
             TopicPartition(message.topic(), message.partition()),
             message.offset(),
             message.value(),
         )
 
+        self.__offsets[result.stream] = result.offset + 1
+
+        return result
+
+    def tell(self) -> Mapping[TopicPartition, int]:
+        if self.__state is ConsumerState.CLOSED:
+            raise RuntimeError("consumer already closed")
+
+        return self.__offsets
+
+    def __seek(self, offsets: Mapping[TopicPartition, int]) -> None:
+        if self.__state is ConsumerState.ASSIGNING:
+            # Trying to use ``seek`` while in an assignment callback will yield
+            # throw an "Erroneous state" error, so seeking actually happens via
+            # assignment in the callback.
+            self.__consumer.assign(
+                [
+                    ConfluentTopicPartition(stream.topic, stream.partition, offset)
+                    for stream, offset in offsets.items()
+                ]
+            )
+            self.__offsets.update(offsets)
```

Comment:
This could probably be moved outside of these conditional blocks since the same thing (in essence) exists on line 207 below.